contrib/storage-hive/core/src/main/codegen/data/HiveFormats.tdd

```diff
@@ -0,0 +1,50 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http:# www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{
```

Comment:
Can we explain this a bit? We have 6 reader types. But, the only difference in generated code is has header/footer or not. Can we solve the Java optimization problem with a "classic" type hierarchy:

```
HiveAbstractReader
. HiveSimpleReader
. . HiveAvroReader
. . ...
. HiveHeaderFooterReader
. . HiveTextReader
. . ...
```

Names are just made up. The point is, can a much simpler Java hierarchy, with less duplicated code, solve the problem? If there is one function that is sub-optimized, can just that one function be generated in the subclass rather than generating duplicate code?